src/Extension.Tests/WorkflowTests.cs

```diff
@@ -77,7 +75,7 @@ public async Task Test()
             await lesson.StartChallengeAsync(challenge1);
```

Comment:
same here, starting a challenge that is not part of a lesson should not be allowed